CMakeLists.txt

```diff
@@ -239,20 +239,20 @@ endmacro()
 # Enabling all warnings in MSVC spams too much
 if(NOT MSVC)
 	add_definitions(-Wall)
-	add_definitions(-D_DEFAULT_SOURCE)
+	add_definitions(-Werror)
 	# TODO: would like these but they produce overwhelming amounts of warnings
 	#check_and_add_flag(EXTRA -Wextra)
-	#check_and_add_flag(MISSING_FIELD_INITIALIZERS -Wmissing-field-initializers)
 	#check_and_add_flag(SWITCH_DEFAULT -Wswitch-default)
 	#check_and_add_flag(FLOAT_EQUAL -Wfloat-equal)
 	#check_and_add_flag(CONVERSION -Wconversion)
 	#check_and_add_flag(ZERO_AS_NULL_POINTER_CONSTANT -Wzero-as-null-pointer-constant)
+	#check_and_add_flag(SHADOW -Wshadow)
```

Comment:
patterns like constructor() : channels(channels) {} keep showing up in the code, which are good for readability but trigger this warning